lib/maths/CBoostedTreeImpl.cc

```diff
@@ -1510,15 +1815,19 @@ std::size_t CBoostedTreeImpl::numberHyperparametersToTune() const {
            (m_FeatureBagFractionOverride != boost::none ? 0 : 1);
 }
 
-std::size_t CBoostedTreeImpl::maximumTreeSize(const core::CPackedBitVector& trainingRowMask) const {
-    return this->maximumTreeSize(static_cast<std::size_t>(trainingRowMask.manhattan()));
+std::size_t CBoostedTreeImpl::maximumTreeSize(const core::CPackedBitVector& trainingRowMask) {
+    return maximumTreeSize(static_cast<std::size_t>(trainingRowMask.manhattan()));
 }
 
-std::size_t CBoostedTreeImpl::maximumTreeSize(std::size_t numberRows) const {
+std::size_t CBoostedTreeImpl::maximumTreeSize(std::size_t numberRows) {
     return static_cast<std::size_t>(
         std::ceil(10.0 * std::sqrt(static_cast<double>(numberRows))));
 }
 
+std::size_t CBoostedTreeImpl::numberTreesToRetrain() const {
+    return m_TreesToRetrain.size();
+}
+
 void CBoostedTreeImpl::recordHyperparameters() {
     m_Instrumentation->hyperparameters().s_Eta = m_Eta;
     m_Instrumentation->hyperparameters().s_ClassAssignmentObjective = m_ClassAssignmentObjective;
```

Comment:
I think, the tree topology change regularization hyperparameter should be added here. As well as the prediction change cost hyperparameter

Reply:
Also I wasn't actually recording hyperparameters in the loop. Fixed in [4c96cfc](https://github.com/elastic/ml-cpp/pull/1837/commits/4c96cfc59912e06b101a4b3d55d025edf090c385).